metrics/storage/webtooling.sh

```diff
@@ -74,55 +131,78 @@ function main() {
 		exit 1
 	fi
 
-	containers=()
-	# Check tools/commands dependencies
-	local cmds=()
-	cmds+=("docker")
+	local i=0
+	local containers=()
+	local cmds=("docker")
+	local not_started_count=$NUM_CONTAINERS
 
 	restart_containerd_service
+	# Check tools/commands dependencies
+	clean
 	init_env
 	check_cmds "${cmds[@]}"
 	check_ctr_images "$IMAGE" "$DOCKERFILE"
-
 	metrics_json_init
-
 	save_config
 
+	info "Creating $NUM_CONTAINERS containers"
+
 	for ((i=1; i<= "$NUM_CONTAINERS"; i++)); do
-		containers+=($(random_name))
+		c_name="$(random_name)"
+		containers+=("$c_name")
 		# Web tool benchmark needs 2 cpus to run completely in its cpu utilization
-		sudo -E ctr run -d --runtime "${CTR_RUNTIME}" --cpu-quota "${cpu_quota}" --cpu-period "${cpu_period}" "$IMAGE" "${containers[-1]}" sh -c "$PAYLOAD_ARGS"
+		sudo -E ctr run -d --runtime "${CTR_RUNTIME}" --cpu-quota "${cpu_quota}" --cpu-period "${cpu_period}" "$IMAGE" "${c_name}" sh -c "$PAYLOAD_ARGS"
+		((not_started_count--))
+		info "$not_started_count remaining containers"
 	done
 
-	# We verify that number of containers that we selected
-	# are running
-	for i in $(seq "$timeout") ; do
-		echo "Verify that the containers are running"
-		containers_launched=$(sudo ctr c list -q | wc -l)
-		[ "$containers_launched" -eq "$NUM_CONTAINERS" ] && break
-		sleep 1
-		[ "$i" == "$timeout" ] && return 1
-	done
+	# Check that the requested number of containers are running
+	local timeout_launch="10"
+	check_containers_are_up & pid=$!
+	(sleep "$timeout_launch" && kill -HUP $pid) 2>/dev/null & pid_tout=$! 
+
+	if wait $pid 2>/dev/null; then
+		pkill -HUP -P $pid_tout
+		wait $pid_tout
+	else
+		warn "Time out exceeded"
+		return 1
+	fi
+
+	# Get the initial number of pids in a single container before the workload starts
+        INITIAL_NUM_PIDS=$(sudo ctr t metrics "${containers[-1]}" | grep pids.current | grep pids.current | xargs | cut -d ' ' -f 2)
+	((INITIAL_NUM_PIDS++))
 
 	# Launch webtooling benchmark
-	CONTAINERS_ID=($(sudo ctr c list -q))
-	for i in "${CONTAINERS_ID[@]}"; do
-		sudo ctr t exec -d --exec-id "$(random_name)" "$i" sh -c "$CMD"
+	local pids=()
+	local j=0
+	for i in "${containers[@]}"; do
+		$(sudo ctr t exec -d --exec-id "$i" "$i" sh -c "$CMD") &
```

Comment:
this is kind of weird `sudo ctr t exec -d --exec-id "$i" "$i"` you are using twice `$i` the variable , I prefer as it was before as it is easier to read

Reply:
use RANDOM instead of two $i

Reply:
@GabyCT the purpose to use the same variable to name both the pid and the container, was because you can avoid an additional random generation call.